estuary_updater/handlers/freshmaker.py

```diff
@@ -79,13 +80,58 @@ def build_state_handler(self, msg):
         """
         build_info = msg['body']['msg']
         event_id = msg['body']['msg']['event_id']
-        build = self.create_or_update_build(build_info, event_id)
-        if build:
-            event = FreshmakerEvent.nodes.get_or_none(id_=str(event_id))
-            if event:
-                event.triggered_container_builds.connect(build)
-            else:
-                log.warn('The Freshmaker event {0} does not exist in Neo4j'.format(event_id))
+        freshmaker_build = None
+        build = None
+        # build_id in Freshmaker is actually the task_id
+        if not build_info['build_id']:
+            log.debug('Skipping build update for event {0} because build_id is not set'.format(
+                event_id))
+            freshmaker_build = self.create_or_update_freshmaker_build(build_info, event_id)
+        # Ignore Freshmaker dry run mode, indicated by a negative ID
+        elif build_info['build_id'] < 0:
+            log.debug('Skipping build update for event {0} because it is a dry run'.format(
+                event_id))
+        else:
+            freshmaker_build = self.create_or_update_freshmaker_build(build_info, event_id)
+            build = self.create_or_update_build(build_info, event_id)
+        event = FreshmakerEvent.nodes.get_or_none(id_=str(event_id))
+        if event:
+            if build:
+                event.successful_koji_builds.connect(build)
+            if freshmaker_build:
+                freshmaker_build.event.conditional_connect(event)
+        else:
+            log.warn('The Freshmaker event {0} does not exist in Neo4j'.format(event_id))
```

Comment:
Optional: I know this predates your PR, but we should change this to `log.warning`